### ISSUE TITLE
False positive type-mismatch when sequentially destructuring a typed map value

### ISSUE THREAD

**Issue by borkdude** [clj-kondo/clj-kondo#2932]
**version**

master at `a8c538e7`, since v2026.07.24.

**problem**

When a map-destructured value with a known type is destructured further with a sequential binding form, the inner bindings get the type of the whole value. Element bindings of a vector value are typed as `vector`, which causes false `:type-mismatch` warnings.

**repro**

```clojure
(let [{[x0 x1] :x} {:x [1 2]}] (+ x0 x1))
```

```
<stdin>:1:33: warning: Expected: number, received: vector.
<stdin>:1:36: warning: Expected: number, received: vector.
```

Also through function return maps:

```clojure
(defn f [] {:x [1 2]})
(let [{[x0 x1] :x} (f)] (+ x0 x1))
```

Nested map destructuring is fine, `(let [{{:keys [a]} :x} {:x {:a 1}}] (inc a))` reports nothing.

**cause**

The `:vector` branch of `extract-bindings` passes `opts` with the value's `:tag` to its children, so each element binding inherits it. `extract-map-bindings` dissocs `:tag` with the comment ":tag must not leak to children wholesale". The `:vector` branch needs the same guard.

**expected behavior**

No warning. Element bindings of a sequentially destructured value have unknown types until per-element inference exists.
